test/test_schemas.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of sample-avro-alert.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os.path
+import unittest
+
+import json
+
+from lsst.alert.packet import get_schema_root, SchemaRegistry
+
+
+def path_to_sample_data(schema_root, version, filename):
+    return os.path.join(schema_root, *version.split("."),
+                        "sample_data", filename)
+
+
+class SchemaValidityTestCase(unittest.TestCase):
+    def setUp(self):
+        self.registry = SchemaRegistry.from_filesystem()
+
+    def test_example_json(self):
+        for version in self.registry.known_versions:
+            path = path_to_sample_data(get_schema_root(), version, "alert.json")
+            if os.path.exists(path):
```

Comment:
What if the path doesn't exist? Shouldn't that be a test failure?

Reply:
So my first answer to this is “no”: this test is intended to take advantage of the sample data where it happens to exist, rather than asserting that it always must exist (because it doesn't). I wonder if it would be worth generating additional sample packets so that *does* always exist, though. I'll see how plausible that is. (I'm a bit worried that the old v1 schema might not be capable of generating valid packets, in which case that'd be doomed to failure.)

Reply:
On reflection, my second answer is “no” as well. :-)

We know that some of the old schemas are just invalid (hence [DM-23588|https://jira.lsstcorp.org/browse/DM-23588]). Worse, we know that some versions of fastavro can handle them, and others will choke. Given that, I don't think it's worthwhile to generate test data which may or may not actually be parseable.

However, I will hard-code a list of versions where we know that test data isn't available, and fail if data isn't available for a schema not on that list, rather than simply relying on the filesystem.